### PR TITLE
test(no-dead-link): Add local file test

### PR DESCRIPTION
**Pull request by azu**
- Update to textlint@10.0.0 and tester 4.0.0
- Use `inputPath` for testing
- Remove unused `checkRelative` option

### CODE REVIEW

File: src/no-dead-link.js [hunk 1]
```diff
@@ -133,8 +131,7 @@ function reporter(context, options = {}) {
 
     if (isRelative(uri)) {
       const filePath = getFilePath();
-      const base = opts.baseURI || (filePath && path.dirname(filePath));
```
Comment (azu) [inline src/no-dead-link.js]:
```
$ node
> var filePath = "/user/path/file.md"
undefined
> require("url").resolve(filePath, "b.md")
'/user/path/b.md'
> require("path").resolve(filePath, "b.md")
'/user/path/file.md/b.md'
> require("url").resolve(require("path").dirname(filePath), "b.md")
'/user/b.md'
```

File: src/no-dead-link.js [hunk 2]
```diff
@@ -1,11 +1,9 @@
 import { RuleHelper } from 'textlint-rule-helper';
 import fetch from 'isomorphic-fetch';
 import URL from 'url';
-import path from 'path';
 import fs from 'fs-extra';
 
 const DEFAULT_OPTIONS = {
-  checkRelative: false, // `true` enables availability checks for relative URIs.
```
Comment (azu) [inline src/no-dead-link.js]: Is this unsed?

File: src/no-dead-link.js (hunk 1 above)
Comment (nodaguti) [reply src/no-dead-link.js]: I haven't noticed there are the differences between `url.resolve` and `path.resolve`.
Thank you for digging the problem!

File: src/no-dead-link.js (hunk 2 above)
Comment (nodaguti) [reply src/no-dead-link.js]: Yes, I've forgot to remove it :P

**Comment by nodaguti** [comment]
Thanks a lot.
You did really great jobs! 💯

**Pull request merged**